### DISCUSSION TITLE
HTML and JavaScript

### DISCUSSION THREAD

**Discussion by manuelstonex** [vagran/dxf-viewer#108] [General]
I need help with running the dxf-viewer (https://github.com/vagran/dxf-viewer/). The available version to run is written in vue.js, which I am not familiar with. I would like to run a simple version using only HTML and JavaScript to parse the dxf file located at the source path: /path/test.dxf. How can I achieve this?

**Comment by vagran** [comment]
See this: https://github.com/vagran/dxf-viewer-example-src/issues/19
Actually, nothing prevents you from doing that, make a bundle and use `dxf-viewer` API. Vue example still can be use as example of `dxf-viewer` API usage.

**Reply by manuelstonex** [reply]
sorry if I bother you, could you give me an example with a predefined dxf file, also because I struggle to understand how the methods should be called

**Reply by vagran** [reply]
There is basically two steps:
Initialize with your canvas: [code](https://github.com/vagran/dxf-viewer-example-src/blob/07737e4def00a328e8c2f8626f10aa15b07354d7/src/components/DxfViewer.vue#L125)
Load the DXF: [code](https://github.com/vagran/dxf-viewer-example-src/blob/07737e4def00a328e8c2f8626f10aa15b07354d7/src/components/DxfViewer.vue#L76)

Web-worker may be used to utilize UI-offloading feature to make UI responsive while processing the DXF. Simple worker script like [this](https://github.com/vagran/dxf-viewer-example-src/blob/07737e4def00a328e8c2f8626f10aa15b07354d7/src/components/DxfViewerWorker.js#L1) is created. The example [uses](https://github.com/vagran/dxf-viewer-example-src/blob/07737e4def00a328e8c2f8626f10aa15b07354d7/src/components/DxfViewer.vue#L17) WebPack worker loader, however in plain JS the factory function could be something like `() => new Worker("DxfViewerWorker.js")`. You may omit the web worker factory argument for the sake of simplicity, then it will process the file synchronously.

I know, it would be nice to have a plain JS example somewhere, but I did not yet manage to find time for that.
